src/engine/structures.rs

```diff
@@ -433,22 +419,22 @@ mod tests {
         let name2 = "name2";
         let thing2 = TestThing::new(&name2, uuid);
         let thing_key2 = thing2.stuff;
-        let displaced = t.insert(thing2);
+        let displaced = t.insert(Name::new(name2.to_owned()), uuid, thing2);
         table_invariant(&t);
 
         // The items displaced consist exactly of the first item.
         assert!(displaced.is_some());
         let ref displaced_item = displaced.unwrap();
-        assert_eq!(&*displaced_item.name(), name);
-        assert_eq!(displaced_item.uuid(), uuid);
-        assert_eq!(displaced_item.stuff, thing_key);
+        assert!(&*displaced_item.0 == name);
+        assert!(displaced_item.1 == uuid);
+        assert!(displaced_item.2.stuff == thing_key);
 
         // The table contains the new item and has no memory of the old.
         assert!(t.contains_uuid(uuid));
         assert!(t.contains_name(name2));
         assert!(!t.contains_name(name));
-        assert_eq!(t.get_by_uuid(uuid).unwrap().stuff, thing_key2);
-        assert_eq!(t.get_by_name(&name2).unwrap().stuff, thing_key2);
-        assert_eq!(t.len(), 1);
+        assert!(t.get_by_uuid(uuid).unwrap().1.stuff == thing_key2);
+        assert!(t.get_by_name(&name2).unwrap().1.stuff == thing_key2);
+        assert!(t.len() == 1);
```

Comment:
Please use `assert_eq` for all of the asserts you are doing which do an equality comparison, which appears to be many of the lines you edited in this commit.

Reply:
Not that I can see. Things must've gotten flipped during some rebasing session.